_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp

```diff
@@ -37,18 +57,23 @@ mfxStatus MFXVideoENCODEVP9_HW::Query(VideoCORE *core, mfxVideoParam *in, mfxVid
     }
     else
     {
+        eMFXHWType platform = core->GetHWType();
+
         // check attached buffers (all are supported by encoder, no duplications, etc.)
         mfxStatus sts = CheckExtBufferHeaders(in->NumExtParam, in->ExtParam);
         MFX_CHECK_STS(sts);
         sts = CheckExtBufferHeaders(out->NumExtParam, out->ExtParam);
         MFX_CHECK_STS(sts);
 
-        VP9MfxVideoParam toValidate = *in;
+        VP9MfxVideoParam toValidate(*in);
+        SetDefaultsForProfileAndFrameInfo(toValidate);
+        SetDefaultForLowpower(toValidate, platform);
```

Comment:
this function returns status. you have to check it

Reply:
Here value of lowpower is checked (that it's set to on/off) to get correct guid; input video parameters isn't changed here and they're corrected below in CheckParametersAndSetDefaults()
So returned status from SetDefaultForLowpower isn't used later, maybe I don't need to return it from this function.

Reply:
If you decide to use return statuses you have to check it at each function call. So you have to do one of the things - check the status or remove return status from function